util/bloom_test.cc

```diff
@@ -458,6 +473,172 @@ TEST_F(FullBloomTest, Schema) {
   ResetPolicy();
 }
 
+struct RawFilterTester {
+  std::array<char, 3000> data_;
+  Slice cur_slice_;
+  char* metadata_ptr_;
+
+  RawFilterTester() : metadata_ptr_(&data_.back() - 4) {}
+
+  Slice& ResetNoFill(uint32_t len_without_metadata, uint32_t num_lines,
+                     uint32_t num_probes) {
+    metadata_ptr_[0] = static_cast<char>(num_probes);
+    EncodeFixed32(metadata_ptr_ + 1, num_lines);
```

Comment:
metadata_ptr_ is &data_.back() - 4. How can we pack fixed32 to metadata_ptr_ + 1? I must have missed something.

Reply:
back is the last byte, not the end. I'll use &*(end() - 5) instead.